lib/mix/tasks/docs.ex

```diff
@@ -141,8 +140,26 @@ defmodule Mix.Tasks.Docs do
     end
   end
 
-  defp normalize_source_beam(options) do
-    Keyword.put_new(options, :source_beam, Mix.Project.compile_path)
+  defp normalize_source_beam(options, config) do
+    compile_path =
+      if Mix.Project.umbrella?(config) do
+        umbrella_compile_paths()
+      else
+        Mix.Project.compile_path
+      end
+
+    Keyword.put_new(options, :source_beam, compile_path)
+  end
+
+  defp umbrella_compile_paths do
```

Comment:
thanks, this is way better! Fixed in e3debf00352fb7e9b4c326e75298c4ad81015c86 where I also added some docs for running within an umbrella.